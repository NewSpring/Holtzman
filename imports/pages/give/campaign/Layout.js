// @flow
/* eslint-disable react/no-danger */
// $FlowMeteor
import { find } from "ramda";
import { Link } from "react-router";

import Split, { Left, Right } from "../../../components/@primitives/layout/split";
import Meta from "../../../components/shared/meta";
import AddToCart from "../../../components/giving/add-to-cart";

type ILayout = {
  account: Object,
};

const Layout = ({ account }: ILayout) => (
  <div>
    <Split nav classes={["background--light-primary"]}>
      <Meta
        title={account.name}
        description={account.summary}
        image={
          account.images ? find(account.images, { fileLabel: "2:1" }).cloudfront : account.image
        }
        meta={[
          { property: "og:type", content: "article" },
        ]}
      />

      <Right
        background={
          account.images ?
            find(account.images, { fileLabel: "2:1" }).cloudfront :
            account.image
        }
        mobile
      />
      <Right
        background={
          account.images ?
            find(account.images, { fileLabel: "1:2" }).cloudfront :
            account.image
        }
        mobile={false}
      />

    </Split>
    <Left scroll classes={["background--light-primary"]}>
      <Link
        to="/give/now"
        className={
          "locked-top locked-left soft-double@lap-and-up soft " +
          "h7 text-dark-secondary plain visuallyhidden@handheld"
        }
      >
        <i
          className="icon-arrow-back soft-half-right display-inline-block"
          style={{ verticalAlign: "middle" }}
        />
        <span
          className="display-inline-block"
          style={{ verticalAlign: "middle", marginTop: "5px" }}
        >
          Back
        </span>
      </Link>
      <div className="soft@lap-and-up soft-double-top@lap-and-up">
        <div className="soft soft-double-bottom soft-double-top@lap-and-up">

          <h2>{account.name}</h2>
          <div dangerouslySetInnerHTML={{ __html: account.description }} />
        </div>
      </div>

      <div className="background--light-secondary">
        <div className="constrain-copy soft-double@lap-and-up">
          <div className="soft soft-double-bottom soft-double-top@lap-and-up">
            <AddToCart accounts={[account]} donate />
          </div>
        </div>
      </div>

    </Left>
  </div>
);

export default Layout;
