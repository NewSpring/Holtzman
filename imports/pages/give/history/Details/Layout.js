/* eslint-disable react/no-danger */
import { Component, PropTypes } from "react";
import moment from "moment";
import { Link } from "react-router";

import Split, { Left, Right } from "../../../../blocks/split";

import { Spinner } from "../../../../components/loading";
import Currency from "../../../../components/currency";
import SideBySide from "../../../../components/cards/SideBySide";
import Meta from "../../../../components/meta";
import AccountType from "../../../../components/accountType";

export default class Layout extends Component {

  static propTypes = {
    loadingEntries: PropTypes.bool,
    entries: PropTypes.array,
    transaction: PropTypes.object,
  }

  formatDate = (date) => (
    moment(new Date(date)).format("MMM D, YYYY")
  )

  /* eslint-disable max-len */
  render() {
    const { loadingEntries, entries } = this.props;
    return (
      <div>
        <Split nav classes={["background--light-primary"]}>
          <Meta title="Giving History" />
          <Right
            background="//dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/all/heroes/newspring/campuses/Florence.1.2x1_1700_850_90_c1.jpg"
            mobile={false}
          />
        </Split>
        <Left scroll ref="container" classes={["background--light-secondary"]}>
          <div
            className={
              "soft-double-sides@lap-and-up soft-double-ends@lap-and-up " +
              "soft background--light-primary"
            }
          >
            {(() => {
              if (process.env.WEB) {
                return (
                  <Link
                    to="/give/history"
                    className={
                      "locked-left soft-double@lap-and-up " +
                      "soft h7 text-dark-secondary plain"
                    }
                  >
                    <i
                      className="icon-arrow-back soft-half-right display-inline-block"
                      style={{ verticalAlign: "middle" }}
                    />
                    <span
                      className="display-inline-block"
                      style={{ verticalAlign: "middle", marginTop: "5px" }}
                    >
                      Back
                    </span>
                  </Link>
                );
              }
              return null;
            })()}
            <div
              className={
                "text-left soft-double-top hard-left@lap-and-up " +
                "soft-half-bottom soft@anchored"
              }
            >
              <div className="soft-double-ends@anchored">
                {(() => {
                  const { transaction } = this.props;
                  if (!transaction) {
                    // loading
                    return (
                      <div className="text-center soft">
                        <Spinner styles={{ width: "40px", height: "40px" }} />
                      </div>
                    );
                  }
                  const { account } = transaction.details[0];
                  const { person } = transaction;
                  return (
                    <div className="text-center">
                      <p className="push-half-bottom">
                        <em>{this.formatDate(transaction.date)}</em>
                      </p>
                      <h3 className="text-dark-secondary">{account.name}</h3>

                      <Currency
                        amount={transaction.details[0].amount.toFixed(2)}
                        baseHeadingSize="1"
                        className="display-inline-block text-center soft-bottom"
                      />

                      <h6 className="push-bottom text-dark-tertiary">
                        {person.nickName || person.firstName} {person.lastName}
                      </h6>

                      {(() => {
                        const detail = transaction.payment;
                        if (detail && detail.accountNumber) {
                          return (
                            <h4 className="text-dark-secondary">
                              {detail.accountNumber.slice(-4)}&nbsp;

                              {(() => {
                                if (detail.paymentType && detail.paymentType === "ACH") {
                                  return (
                                    <AccountType width="30px" height="20px" type="Bank" />
                                  );
                                } else if (detail.paymentType) {
                                  return (
                                    <AccountType
                                      width="30px"
                                      height="20px"
                                      type={detail.paymentType}
                                    />
                                  );
                                }
                                return null;
                              })()}

                            </h4>
                          );
                        }
                        return null;
                      })()}
                      <p
                        className={
                          "text-center soft-ends soft-double@anchored " +
                          "flush-bottom soft-ends soft-sides@portable"
                        }
                      >
                        Thank you so much for your contributions! It is because of your generosity that we are able to continue telling stories of the greatness of Jesus and seeing lives changed.
                      </p>
                    </div>
                  );
                })()}
              </div>
            </div>
          </div>
          <div className="soft-half soft-sides@portable soft-double-sides@anchored">

            <h4 className="soft soft-double-ends text-center@lap-and-up flush-bottom">
              Recent Articles About Giving
            </h4>
            {(() => {
              if (loadingEntries) {
                return <div className="one-whole soft text-center"><Spinner /></div>;
              }
              return (
                <div className="grid">
                  {entries && entries.map((entry, key) => (
                    <div
                      key={key}
                      className={
                        "grid__item one-whole push-half-bottom push-bottom@portable hard-bottom"
                      }
                    >
                      <SideBySide
                        classes={["push-bottom@lap-and-up"]}
                        images={entry.content.images}
                        defaultImage={entry.content.images[0].url}
                      >
                        <h4 className="push-half-top@portable push-top@anchored">
                          {entry.title}
                        </h4>

                        <p><small dangerouslySetInnerHTML={{ __html: entry.meta.summary }} /></p>
                        {(() => {
                          if (process.env.WEB) {
                            return (
                              <a
                                rel="noopener noreferrer"
                                target="_blank"
                                href={`https://newspring.cc/articles/${entry.meta.urlTitle}`}
                                className={
                                  "h6 btn--small btn--dark-tertiary " +
                                  "soft-sides@portable one-whole@handheld"
                                }
                              >
                                Read more
                              </a>
                            );
                          }

                          if (process.env.NATIVE) {
                            return (
                              <Link
                                to={`/articles/${entry.entryId}`}
                                className={
                                  "h6 btn--small btn--dark-tertiary " +
                                  "soft-sides@portable one-whole@handheld"
                                }
                              >
                                Read more
                              </Link>
                            );
                          }

                          return null;
                        })()}


                      </SideBySide>
                    </div>
                  ))}
                </div>
              );
            })()}

          </div>
        </Left>
      </div>

    );
  }
  /* eslint-enable max-len */
}
