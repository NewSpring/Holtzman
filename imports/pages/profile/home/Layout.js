import { PropTypes } from "react";
import { Meteor } from "meteor/meteor";
import Meta from "../../../components/meta";

import Split, { Left, Right } from "../../../blocks/split";
import Toggle from "../../../components/controls/toggle";


const Layout = ({ photo, person, onToggle, content, onUpload }) => (
  <div>
    <Split nav classes={["background--light-primary"]}>

      <Meta
        title={person ? `${person.nickName} ${person.lastName}` : "Sign In"}
      />

      <Right
        mobile
        classes={["floating", "overlay--solid-dark"]}
        ratioClasses={["floating__item", "overlay__item", "one-whole", "text-center"]}
        background={photo}
        blur
      >
        <div className="soft one-whole">
          <label
            htmlFor="file"
            className="background--fill ratio--square round two-fifths display-inline-block"
            style={{ backgroundImage: `url('${photo}')`, position: "relative" }}
          >
            {(() => {
              if (!Meteor.isCordova) {
                return (
                  <input onChange={onUpload} type="file" className="locked-ends locked-sides" style={{ opacity: 0 }} />
                );
              }

              return (
                <div onClick={onUpload} className="locked-ends locked-sides" style={{ opacity: 0 }} />
              );
            })()}
          </label>
          <h4
            className="text-light-primary soft-half-top flush-bottom"
          >
            {person.nickName} {person.lastName}
          </h4>
          {(() => {
            if (!person.home || !person.home.city) return null;
            return (
              <p className="text-light-primary flush"><em>{person.home.city}</em></p>
            );
          })()}
        </div>

      </Right>


    </Split>
    <Left scroll>
      <Toggle items={["Likes", "Following"]} toggle={onToggle} />
      <div>
        {content}
      </div>
    </Left>
  </div>

);

Layout.propTypes = {
  photo: PropTypes.object.isRequired,
  person: PropTypes.object.isRequired,
  onToggle: PropTypes.func.isRequired,
  content: PropTypes.object.isRequired,
  onUpload: PropTypes.func.isRequired,
};

export default Layout;
