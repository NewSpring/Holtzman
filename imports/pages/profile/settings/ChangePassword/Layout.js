import PropTypes from 'prop-types';
import Forms from "../../../../components/@primitives/UI/forms";
import Meta from "../../../../components/shared/meta";

import Back from "../Back";

const styles = {
  overflow: "visible",
  zIndex: 1,
};

const Layout = ({ submit, save, state }) => (
  <div
    className={
      "background--light-primary one-whole text-center " +
      "push-top push-double-top@lap-and-up push-double-bottom"
    }
    style={styles}
  >
    <Back />
    <Meta title="Change your password" />
    <Forms.Form
      id="reset-password"
      classes={[
        "soft",
        "one-whole",
        "two-thirds@portable",
        "one-half@anchored",
        "display-inline-block",
        "soft-double-sides@palm-wide",
      ]}
      submit={submit}
    >
      <div className="push-double">
        <h4 className="text-center">
          Change Password
        </h4>
      </div>

      <Forms.Input
        name="current"
        label="Current Password"
        validation={save}
        type="password"
      />

      <Forms.Input
        name="newP"
        label="New Password"
        validation={save}
        errorText="New password does not match"
        type="password"
      />

      <Forms.Input
        name="newPDup"
        label="Repeat New Password"
        validation={save}
        errorText="New password does not match"
        type="password"
      />
      {(() => {
        const btnClasses = [];
        let disabled = true;
        const { current, newP, newPDup } = state;
        if (!current || !newP || !newPDup) {
          btnClasses.push("btn--disabled");
        } else {
          btnClasses.push("btn");
          disabled = false;
        }

        return (
          <button className={btnClasses.join(" ")} disabled={disabled}>
            Enter
          </button>
        );
      })()}
    </Forms.Form>
  </div>
);

Layout.propTypes = {
  submit: PropTypes.func.isRequired,
  save: PropTypes.func.isRequired,
  state: PropTypes.object.isRequired,
};

export default Layout;
