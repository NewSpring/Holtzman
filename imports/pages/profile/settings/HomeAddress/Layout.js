import { Component, PropTypes } from "react";

import Meta from "../../../../components/meta";
import Forms from "../../../../components/forms";

import Back from "../Back";

export default class Layout extends Component {
  static propTypes = {
    home: PropTypes.object,
    update: PropTypes.func,
  }

  submit = (e) => {
    e.preventDefault();

    const data = {};
    // eslint-disable-next-line
    for (const ref in this.refs) {
      let value = this.refs[ref].getValue();
      const number = Number(value);
      if (number) {
        value = number;
      }

      data[ref] = value;
    }

    this.props.update(data);
  }

  render() {
    const { home } = this.props;

    const {
      city,
      zip,
      state,
      street1,
      street2,
    } = home;

    return (
      <div
        className={
          "background--light-primary one-whole text-center push-top " +
          "push-double-top@lap-and-up push-double-bottom background--light-primary"
        }
      >
        <Meta title="Change your address" />
        <Back />
        <Forms.Form
          id="reset-password"
          classes={[
            "soft",
            "one-whole",
            "two-thirds@portable",
            "one-half@anchored",
            "display-inline-block",
            "soft-double-sides@palm-wide",
          ]}
          submit={this.submit}
        >
          <div className="push-double">
            <h4 className="text-center">
              My Home Address
            </h4>
          </div>

          <Forms.Input
            name="Street1"
            label="Street"
            ref="Street1"
            type="text"
            defaultValue={street1}
          />
          <Forms.Input
            name="Street2"
            label="Street 2 (Optional)"
            ref="Street2"
            type="text"
            defaultValue={street2}
          />
          <div className="grid">

            <div className="grid__item two-fifths">
              <Forms.Input
                name="City"
                label="City"
                defaultValue={city}
                ref="City"
              />
            </div>

            <div className="grid__item three-fifths">

              <div className="grid">

                <div className="grid__item one-half">
                  <Forms.Input
                    name="State"
                    label="State"
                    defaultValue={state}
                    ref="State"
                  />

                </div>
                <div className="grid__item one-half">
                  <Forms.Input
                    name="PostalCode"
                    label="Zip"
                    defaultValue={zip}
                    ref="PostalCode"
                  />
                </div>
              </div>
            </div>
          </div>

          {(() => {
            const btnClasses = [];
            const ready = true;
            if (!ready) {
              btnClasses.push("btn--disabled");
            } else {
              btnClasses.push("btn");
            }

            return (
              <button className={btnClasses.join(" ")}>
                Update
              </button>
            );
          })()}


        </Forms.Form>
      </div>
    );
  }
}
