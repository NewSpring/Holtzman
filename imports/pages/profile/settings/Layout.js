import { PropTypes } from "react";
import { Link } from "react-router";

import Split, { Left, Right } from "../../../blocks/split";
import Meta from "../../../components/meta";

const Close = () => (
  <Link to="/profile" className="text-light-primary plain soft overlay__item locked-top locked-right">
    <i className="icon-close h4" />
  </Link>
);

const Layout = ({ person, children, mobile, onUpload }) => (
  <div>
    <Meta
      title={person.nickName ? `${person.nickName} ${person.lastName}` : "Profile"}
      image={person.photo ? person.photo : null}
    />
  <Split nav classes={mobile ? ["background--light-secondary"] : ["background--light-primary"]}>
      <Right
        mobile={mobile}
        classes={["floating", "overlay--solid-dark"]}
        ratioClasses={["floating__item", "overlay__item", "one-whole", "text-center"]}
        background={person.photo ? person.photo : null}
        blur
      >
        <div className="soft one-whole">
          <label htmlFor="file"
            className="background--fill ratio--square round two-fifths display-inline-block"
            style={{ backgroundImage: `url('${person.photo ? person.photo : ""}')`, position: "relative" }}
          >
            <input onChange={onUpload} type="file" className="locked-ends locked-sides" style={{ opacity: 0 }} />
          </label>
        <h4 className="text-light-primary soft-half-top flush-bottom">{person.nickName || person.firstName} {person.lastName}</h4>
          <p className="text-light-primary flush"><em>{person.home && person.home.city}, {person.home && person.home.state}</em></p>
        </div>

      </Right>

    </Split>

    <Left scroll classes={["background--light-primary"]}>
      {children}
    </Left>
  </div>

);

Layout.propTypes = {
  photo: PropTypes.string,
  person: PropTypes.object,
};

export default Layout;
