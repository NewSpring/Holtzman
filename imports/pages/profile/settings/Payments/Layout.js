import { PropTypes } from "react";

import Meta from "../../../../components/meta";
import Loading from "../../../../components/loading";

import Back from "../Back";

import SavedPayment from "./SavedPayment";

const styles = {
  overflow: "visible",
  zIndex: 1,
};

const Layout = ({ details, remove, loading }) => (
  <div
    className={
      "background--light-primary text-center soft-double-top " +
      "push-double-bottom push-top push-double-top@lap-and-up soft-double-sides@palm-wide"
    }
    style={styles}
  >
    <Meta title="Saved Payments" />
    <Back />
    <div className="one-whole two-thirds@anchored display-inline-block push-double-top">
      <h3>Saved Accounts</h3>
      <div className="soft-sides soft-double-sides@lap-and-up text-center">
        {loading && <Loading />}
        {!loading && details && details.map((account, key) => (
          <SavedPayment account={account} remove={remove} key={key} />
        ))}
        <p className="soft-ends text-left">
          To add a saved account, click the option to save account on your next gift!
        </p>
      </div>
    </div>
  </div>
);

Layout.propTypes = {
  details: PropTypes.array.isRequired,
  remove: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
};

export default Layout;
