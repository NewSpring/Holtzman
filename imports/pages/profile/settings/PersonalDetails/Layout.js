import PropTypes from 'prop-types';
import { Component } from "react";

import Forms from "../../../../components/@primitives/UI/forms";
import Meta from "../../../../components/shared/meta";
import Validate from "../../../../util/validate";

import Back from "../Back";

export default class Layout extends Component {

  static propTypes = {
    submit: PropTypes.func.isRequired,
    person: PropTypes.object,
    months: PropTypes.array.isRequired,
    saveMonth: PropTypes.func.isRequired,
    days: PropTypes.array.isRequired,
    years: PropTypes.array.isRequired,
    campuses: PropTypes.array,
  }

  submit = e => {
    e.preventDefault();

    const data = {};
    // eslint-disable-next-line
    for (const ref in this.refs) {
      let value = this.refs[ref].getValue();
      if (ref === "Email" && !Validate.isEmail(value)) {
        continue; // eslint-disable-line
      }

      const number = Number(value);
      if (number) {
        value = number;
      }

      data[ref] = value;
    }

    this.props.submit(data);
  }

  render() {
    const { person, months, saveMonth, days, years, campuses } = this.props;

    const {
      campus,
      firstName,
      lastName,
      nickName,
      email,
      birthDay,
      birthMonth,
      birthYear,
    } = person;

    return (
      <div
        className={
          "background--light-primary one-whole text-center push-top " +
          "push-double-top@lap-and-up soft-double-bottom push-double-bottom"
        }
      >
        <Meta title="Update your details" />
        <Back />
        <Forms.Form
          id="reset-password"
          classes={[
            "soft",
            "one-whole",
            "two-thirds@portable",
            "one-half@anchored",
            "display-inline-block",
            "soft-double-sides@palm-wide",
          ]}
          submit={this.submit}
        >
          <div className="push-double">
            <h4 className="text-center">
              My Personal Details
            </h4>
          </div>

          <h6 className="soft-bottom">Name</h6>
          <Forms.Input
            name="NickName"
            label="Nickname"
            ref="NickName"
            type="text"
            defaultValue={nickName}
          />

          <div className="grid">
            <div className="grid__item one-half">

              <Forms.Input
                name="FirstName"
                label="First Name"
                ref="FirstName"
                type="text"
                defaultValue={firstName}
              />
            </div>
            <div className="grid__item one-half">

              <Forms.Input
                name="LastName"
                label="Last Name"
                ref="LastName"
                type="text"
                defaultValue={lastName}
              />
            </div>
          </div>


          <h6 className="soft-bottom">Contact</h6>
          <Forms.Input
            name="Email"
            placeholder="user@email.com"
            label="Email"
            type="email"
            ref="Email"
            errorText="Please enter a valid email"
            validation={Validate.isEmail}
            defaultValue={email}
          />

          <h6 className="soft-bottom">Birthday</h6>
          <div className="grid">
            <div className="grid__item three-fifths">
              <div className="grid">
                <div className="grid__item one-half">
                  <Forms.Select
                    name="BirthMonth"
                    label="Month"
                    ref="BirthMonth"
                    type="text"
                    defaultValue={birthMonth}
                    includeBlank
                    items={months}
                    validation={saveMonth}
                  />
                </div>
                <div className="grid__item one-half">
                  <Forms.Select
                    name="BirthDay"
                    label="Day"
                    ref="BirthDay"
                    type="text"
                    defaultValue={birthDay}
                    includeBlank
                    items={days}
                  />
                </div>

              </div>

            </div>
            <div className="grid__item two-fifths">
              <Forms.Select
                name="BirthYear"
                label="Year"
                ref="BirthYear"
                type="text"
                defaultValue={birthYear}
                includeBlank
                items={years}
              />
            </div>
          </div>
          <h6 className="soft-bottom">Campus</h6>
          <Forms.Select
            name="Campus"
            label="Campus"
            type="Campus"
            defaultValue={(campus && campus.id) || false}
            ref="Campus"
            includeBlank
            items={campuses || []}
          />

          {/*
          <Link
            to="/profile/settings"
            tabIndex={-1}
            className="btn--small btn--dark-tertiary display-inline-block"
          >
            Back
          </Link>
          */}
          {(() => {
            const btnClasses = [];
            const ready = true;
            if (!ready) {
              btnClasses.push("btn--disabled");
            } else {
              btnClasses.push("btn");
            }

            return (
              <button className={btnClasses.join(" ")}>
                Update
              </button>
            );
          })()}
        </Forms.Form>
      </div>
    );
  }
}
