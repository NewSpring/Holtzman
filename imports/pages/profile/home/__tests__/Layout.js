import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { Meteor } from "meteor/meteor";
import Layout from "../Layout";

const defaultProps = {
  photo: "http://test.com/photo.jpg",
  person: {
    nickName: "jim",
    lastName: "bob",
    home: {
      city: "anderson",
    },
  },
  onToggle: jest.fn(),
  content: <h1>test</h1>,
  onUpload: jest.fn(),
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <Layout { ...newProps } />;
};

Meteor.isCordova = false;

it("renders with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders cordova version", () => {
  Meteor.isCordova = true;
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("works without city", () => {
  const wrapper = shallow(generateComponent({
    person: {
      nickName: "jim",
      lastName: "bob",
    },
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
