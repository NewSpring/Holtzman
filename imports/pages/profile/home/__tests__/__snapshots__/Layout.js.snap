exports[`test renders cordova version 1`] = `
<div>
  <Connect(SplitContainer)
    classes={
      Array [
        "background--light-primary",
      ]
    }
    nav={true}>
    <Meta
      title="jim bob" />
    <Right
      background="http://test.com/photo.jpg"
      blur={true}
      classes={
        Array [
          "floating",
          "overlay--solid-dark",
        ]
      }
      mobile={true}
      ratioClasses={
        Array [
          "floating__item",
          "overlay__item",
          "one-whole",
          "text-center",
        ]
      }>
      <div
        className="soft one-whole">
        <label
          className="background--fill ratio--square round two-fifths display-inline-block"
          htmlFor="file"
          style={
            Object {
              "backgroundImage": "url(\'http://test.com/photo.jpg\')",
              "position": "relative",
            }
          }>
          <div
            className="locked-ends locked-sides"
            onClick={[Function]}
            style={
              Object {
                "opacity": 0,
              }
            } />
        </label>
        <h4
          className="text-light-primary soft-half-top flush-bottom">
          jim
           
          bob
        </h4>
        <p
          className="text-light-primary flush">
          <em>
            anderson
          </em>
        </p>
      </div>
    </Right>
  </Connect(SplitContainer)>
  <Left
    scroll={true}>
    <Toggle
      items={
        Array [
          "Likes",
          "Following",
        ]
      }
      toggle={[Function]} />
    <div>
      <h1>
        test
      </h1>
    </div>
  </Left>
</div>
`;

exports[`test renders with props 1`] = `
<div>
  <Connect(SplitContainer)
    classes={
      Array [
        "background--light-primary",
      ]
    }
    nav={true}>
    <Meta
      title="jim bob" />
    <Right
      background="http://test.com/photo.jpg"
      blur={true}
      classes={
        Array [
          "floating",
          "overlay--solid-dark",
        ]
      }
      mobile={true}
      ratioClasses={
        Array [
          "floating__item",
          "overlay__item",
          "one-whole",
          "text-center",
        ]
      }>
      <div
        className="soft one-whole">
        <label
          className="background--fill ratio--square round two-fifths display-inline-block"
          htmlFor="file"
          style={
            Object {
              "backgroundImage": "url(\'http://test.com/photo.jpg\')",
              "position": "relative",
            }
          }>
          <input
            className="locked-ends locked-sides"
            onChange={[Function]}
            style={
              Object {
                "opacity": 0,
              }
            }
            type="file" />
        </label>
        <h4
          className="text-light-primary soft-half-top flush-bottom">
          jim
           
          bob
        </h4>
        <p
          className="text-light-primary flush">
          <em>
            anderson
          </em>
        </p>
      </div>
    </Right>
  </Connect(SplitContainer)>
  <Left
    scroll={true}>
    <Toggle
      items={
        Array [
          "Likes",
          "Following",
        ]
      }
      toggle={[Function]} />
    <div>
      <h1>
        test
      </h1>
    </div>
  </Left>
</div>
`;

exports[`test works without city 1`] = `
<div>
  <Connect(SplitContainer)
    classes={
      Array [
        "background--light-primary",
      ]
    }
    nav={true}>
    <Meta
      title="jim bob" />
    <Right
      background="http://test.com/photo.jpg"
      blur={true}
      classes={
        Array [
          "floating",
          "overlay--solid-dark",
        ]
      }
      mobile={true}
      ratioClasses={
        Array [
          "floating__item",
          "overlay__item",
          "one-whole",
          "text-center",
        ]
      }>
      <div
        className="soft one-whole">
        <label
          className="background--fill ratio--square round two-fifths display-inline-block"
          htmlFor="file"
          style={
            Object {
              "backgroundImage": "url(\'http://test.com/photo.jpg\')",
              "position": "relative",
            }
          }>
          <div
            className="locked-ends locked-sides"
            onClick={[Function]}
            style={
              Object {
                "opacity": 0,
              }
            } />
        </label>
        <h4
          className="text-light-primary soft-half-top flush-bottom">
          jim
           
          bob
        </h4>
      </div>
    </Right>
  </Connect(SplitContainer)>
  <Left
    scroll={true}>
    <Toggle
      items={
        Array [
          "Likes",
          "Following",
        ]
      }
      toggle={[Function]} />
    <div>
      <h1>
        test
      </h1>
    </div>
  </Left>
</div>
`;
